cb-tools/burgbox/+io/Communicator.m

```diff
@@ -3,12 +3,27 @@
   %   Interface encapsulating some connection able to send and receive
   %   messages carrying arbitrary data.
   %
+  %   NB: 2018b and below do not support validation functions in abstract
+  %   property definitions.  Hence we have commented them out.
+  %
   % Part of Burgbox
 
   % 2013-03 CB created
 
   properties (Abstract, SetAccess = protected)
```

Comment:
add comments with descriptions of these properties/methods? I know you'll probably say "they're self-descriptive", but I think it's good practice, and I'm not positive what all of these do (e.g. the `EventMode` property and the `open` method - I assume the latter enables listening, but I'm not certain).

Reply:
Yes this class needs further documenting